### PR TITLE
OLS-58: Limit conversation history

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Limit conversation history

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-58](https://issues.redhat.com//browse/OLS-58)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Comment by onmete** [comment]
So the messages are truncated per message and not in the actual message (like only half of the message is returned), right?
In theory, a single msg (that is long enough) from the user can truncate history to zero msgs? Do we have a limit on the user prompt?

**Comment by tisnik** [comment]
> So the messages are truncated per message and not in the actual message (like only half of the message is returned), right? In theory, a single msg (that is long enough) from the user can truncate history to zero msgs? Do we have a limit on the user prompt?

yes, truncation per message. To not to truncate in middle of sentence or so. User prompt limit is checked in existing code (above this one IIRC).

**Comment by onmete** [comment]
I see `DEFAULT_RESPONSE_TOKEN_LIMIT`, but nothing for prompts. Just thinking if we need to set it too.

File: ols/constants.py [hunk 1]
```diff
@@ -180,6 +180,7 @@
 # model configs
 DEFAULT_CONTEXT_WINDOW_SIZE = 2000
 DEFAULT_RESPONSE_TOKEN_LIMIT = 500
+DEFAULT_HISTORY_TOKEN_LIMIT = 2000
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/constants.py:183]: yup that's correct.

File: ols/src/query_helpers/docs_summarizer.py [hunk 2]
```diff
@@ -87,6 +87,29 @@ def _get_rag_data(self, rag_index: VectorStoreIndex, query: str) -> list[dict]:
 
         return token_handler_obj.truncate_rag_context(retrieved_nodes, available_tokens)
 
+    def limit_conversation_history(
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/query_helpers/docs_summarizer.py:90]: will do

[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/query_helpers/docs_summarizer.py]: it has been moved

File: ols/constants.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/constants.py]: removed

**Pull request merged**